Automattic/newspack-blocks
# Block Styles

### All Submissions:






### Changes proposed in this Pull Request:

This PR sets up a way to define block styles for blocks that are not part of the plugin, e.g. Core blocks, Jetpack blocks, etc. The first example is the core Columns block. To  add styles to a new block, create a directory for it in `src/block-styles` matching the block name (e.g. `core/image` or `jetpack/slideshow`). The directory should contain three files: `index.js`, `editor.scss` and `view.scss`. The index defines the needed blocks styles while the other two style sheets control appearance in editor and view. There's a change to the Webpack config to generate a view stylesheet with block styles in it, which will be enqueued always, not just when certain blocks are present.

### How to test the changes in this Pull Request:

1. Check out the branch, run `npm run build:webpack`
2. Install/Activate Gutenberg plugin, for improved Columns block.
3. Add Columns block, verify the Block Styles are available, and that they make the appropriate changes in Editor and View.

### Other information:
